### DISCUSSION TITLE
Passing Object to Edge Components

### DISCUSSION THREAD

**Discussion by liam-edwards** [adonisjs/core#2473] [Help]
I have a component that accepts two fields, `skill` and `proficiency`. I have created an array of these skill objects and am looping over them to include each component and I'm passing the object through. The problem is when accessing `skill` and `proficiency` in the component, they come up `undefined`.

Here's a snippet of what I have currently.

```
@set('newSkill', (skill, proficiency) => ({ skill, proficiency }))
@set('skills', [
	newSkill('Skill 1', 0.9),
	newSkill('Skill 2', 0.4),
	newSkill('Skill 3', 0.7),
	newSkill('Skill 4', 0.7),
	newSkill('Skill 5', 0.7),
	newSkill('Skill 6', 0.9),
])
@each(skill in skills)
	@!component('components/skill-bar', skill)
@endeach
```

I found that to get it to work, I have to create a new object:

```
@!component('components/skill-bar', { skill: skill.skill, proficiency: skill.proficiency })
```

Is this intended behaviour? Am I missing something? Any help here would be appreciated.

**Comment by thetutlage** [comment]
Can you please share the version of edge you are using + the markup for the component

**Reply by liam-edwards** [reply]
According to my yarn.lock file, I am using v3.1.0 of @adonisjs/view and v3.5.1 of edge.js.

The component (include alpine.js)
```
<div x-data="{ skill: '{{ skill }}', proficiency: {{ proficiency }} }" class="w-40 mb-4">
	<p class="text-sm font-semibold mb-2" x-text="skill"></p>
	<span class="h-2 bg-gray-400 block">
		<span class="h-2 bg-gray-600 dark:bg-purple-900 block" style="width: {{ proficiency * 100 }}%"></span>
	</span>
</div>
```

**Comment by Uttam-EWW** [comment]
I'm facing same issue.
`
@!component('components/table_column', column)
`
where `column` is object from controller. But i'm not able to access it value in component file.
